app/controllers/api/alert_statuses_controller.rb

```diff
@@ -0,0 +1,55 @@
+module Api
+  class AlertStatusesController < BaseController
+    include Subcollections::AlertStatusStates
+    def options
+      payload = {"providers" => []}
+      ExtManagementSystem.select do |m|
+        m.kind_of?(ManageIQ::Providers::CloudManager) || m.kind_of?(ManageIQ::Providers::InfraManager) || m.kind_of?(ManageIQ::Providers::ContainerManager)
+      end.each do |provider|
+        payload["providers"] << {
+          "environment" => "production",
+          "name"        => provider.name,
+          "type"        => provider.class.to_s,
+          "id"          => provider.id}.merge("alerts" => alerts_and_states(provider))
+      end
+      render_options(:alert_statuses, payload)
+    end
```

Comment:
With many providers, this is potentially a lots of data.  Instead of OPTIONS /api/alert_statuses, I would prefer if this information is gathered via a provider subcollection, i.e. GET /api/providers/:id/alert_statuses. If needed for all providers, then it's a GET /api/providers?expand=resources,alert_statuses